### PR TITLE
Implement CSV v2 in host report

### PR DESCRIPTION
**Pull request by znarf**
Part of https://github.com/opencollective/opencollective/issues/4590

### CODE REVIEW

**Review by kewitz — APPROVED** [review]
Awesome. It works perfectly! 💯

File: templates/emails/host.report.hbs [hunk 1]
```diff
@@ -446,8 +446,11 @@ Subject: {{#if erratumMessage}}Erratum{{#if erratumNumber}} {{erratumNumber}}{{/
   <h2>📎 Attachments</h2>
   <ul>
     <li>A CSV export of all the transactions for this {{#if month}}month{{else}}year{{/if}}</li>
+    {{#if csvV2}}
+      <li>NEW: A second CSV export in a different format (v2). Send us your feedback!</li>
```
Comment (SudharakaP) [inline templates/emails/host.report.hbs:450]: Maybe add information on how to send feedback. 🤔 

```suggestion
      <li>NEW: A second CSV export in a different format (v2). Feel free to send us your feedback. Just reply to this email!</li>
```

File: server/lib/csv.js [hunk 2]
```diff
@@ -39,3 +37,19 @@ export const getCollectiveTransactionsCsv = async (collective, { startDate, endD
       logger.error(`Error fetching CSV: ${error.message}`);
     });
 };
+
+export const getCollectiveTransactionsCsv = async (collective, { startDate, endDate } = {}) => {
+  if (parseToBoolean(config.restService.fetchCollectiveTransactionsCsv) === false) {
```
Comment (SudharakaP) [inline server/lib/csv.js:42]: Maybe simplify a bit... 🤔 

```suggestion
  if (!parseToBoolean(config.restService.fetchCollectiveTransactionsCsv)) {
```

File: server/lib/csv.js [hunk 3]
```diff
@@ -39,3 +37,19 @@ export const getCollectiveTransactionsCsv = async (collective, { startDate, endD
...
+    return;
+  }
+
+  return getTransactionsCsv('transactions', collective, { startDate, endDate });
+};
+
+export const getHostTransactionsCsv = async (collective, { startDate, endDate } = {}) => {
+  if (parseToBoolean(config.restService.fetchHostTransactionsCsv) === false) {
```
Comment (SudharakaP) [inline server/lib/csv.js:50]:
```suggestion
  if (!parseToBoolean(config.restService.fetchHostTransactionsCsv)) {
```

File: server/lib/csv.js (hunk 2 above)
Comment (kewitz) [reply server/lib/csv.js:42]: If you're touching this, I would suggest to always test for the condition you want to happen and not the opposite condition:
```ts
if (parseToBoolean(config.restService.fetchCollectiveTransactionsCsv)) {
  return getTransactionsCsv('transactions', collective, { startDate, endDate });
}
```

Comment (znarf) [reply server/lib/csv.js:42]: I like the opposite. At the beginning, try to find all the reasons to not do it, return early or throw exceptions. Then, when you're good, proceed without any condition. No nested code and generally easier to read.

Comment (znarf) [reply server/lib/csv.js:42]: In different cases, generally a good idea to favor positive assertions versus negative one.

**Pull request merged**